src/main/java/uk/nhs/prm/repo/ehrtransferservice/gp2gp_message_models/MessageContent.java

```diff
@@ -1,4 +1,4 @@
 package uk.nhs.prm.repo.ehrtransferservice.gp2gp_message_models;
 
-public abstract class MessageContent {
+public interface MessageContent {
```

Comment:
Marker interfaces are seemingly looked at with a negative light in modern Java, it's likely to point to a code smell! Maybe we can revisit this in the future and take a new architectural approach 😄 

Sources:

https://www.baeldung.com/java-marker-interfaces#:~:text=A%20marker%20interface%20is%20also,we%20should%20use%20them%20carefully.

Reply:
Funnily enough, SonarCloud flagged the marker Abstract Class as a code smell and asked that we change it to an interface! I'm not a fan of them either, I'm sure there is some logic we can move into an abstract class, but the only thing that implements this interface is one of those messy webs of publicly visible fields.

I'll leave it for now, fully agree with your point!